app/views/login/__init__.py

```diff
@@ -37,15 +37,17 @@ def load_user_from_request(request):
     if not token:
         return None
 
-    if not token.user:
-        name = request.headers.get('name')
-        if not name:
-            name = request.args.get('name')
-        if not name:
-            name = "<unnamed>"
-        token.user = User(name=name, rooms=[token.room])
+    name = request.headers.get('name') or request.args.get('name')
+    if not name:
+        name = '<unnamed>'
+
+    user = token.users.filter_by(name=name, session_id=None).one_or_none()
+
+    if not user:
+        user = User(name=name, token=token, rooms=[token.room])
+        db.add(user)
         db.commit()
```

Comment:
As logging in via request is only used for bots, I'd remove the lookup and simply create a new user every time a bot connects. It's not expected, that bots will start several times but only at the very start of an experiment.

```suggestion
    user = User(name=name, token=token, rooms=[token.room])
    db.add(user)
    db.commit()
```

Reply:
Are you sure, that this function is only triggered on connection? For the MinimalBot a logging statement inserted indicates that it is called 2-3 times. The first connection, the emit statement and upon disconnection. If this function is indeed called for every emit, creating a user each time would become problematic.

Reply:
I always assumed this, I'm not entirely sure, however. But if it's actually called multiple times, we need a better solution than searching the user list for a given user name but I can't imagine a simple solution without storing the user id somehow at the client and make different requests based on the current state. Sorry, for the confusion.